subprojects/language-native/src/main/java/org/gradle/language/nativeplatform/tasks/AbstractNativeCompileTask.java

```diff
@@ -146,6 +148,9 @@ protected void configureSpec(NativeCompileSpec spec) {
     private <T extends NativeCompileSpec> WorkResult doCompile(T spec, PlatformToolProvider platformToolProvider) {
         Class<T> specType = Cast.uncheckedCast(spec.getClass());
         Compiler<T> baseCompiler = platformToolProvider.newCompiler(specType);
+        if (baseCompiler instanceof SystemIncludesAwareNativeCompiler) {
+            spec.include(((SystemIncludesAwareNativeCompiler<T>) baseCompiler).getSystemIncludes());
+        }
```

Comment:
From the comment above, we can remove this and rely on the include path being correct.